charts/pure-ftpd/templates/deployment.yaml

```diff
@@ -0,0 +1,96 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Release.Name }}
+    spec:
+      volumes:
+        - name: data
+          persistentVolumeClaim:
+            claimName: {{ .Release.Name }}-data-storage
+        - name: config
+          emptyDir: {}
+      initContainers:
+        # Set up UID/GID on data storage before creating
+        # the corresponding user.
+        - name: set-permissions
+          image: alpine
+          volumeMounts:
+            - name: data
+              mountPath: /data
+          env:
+            {{- with .Values.userSecret }}
+            - name: USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .secretName }}
+                  key: {{ .usernameKey }}
+            {{- end}}
+          command: ["sh", "-c"]
+          args: ["mkdir /data/${USERNAME} && chown -R 1003:1005 /data/${USERNAME}"]
+
+        # Register a virtual user, rooted in the data volume.
+        - name: add-user
+          image: crazymax/pure-ftpd:1.0.49
+          volumeMounts:
+            - name: config
+              mountPath: /etc/pureftpd
+          env:
+            - name: PURE_PASSWDFILE
+              value: /etc/pureftpd/pureftpd.passwd
+            - name: PURE_DBFILE
+              value: /etc/pureftpd/pureftpd.pdb
+            {{- with .Values.userSecret }}
+            - name: USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .secretName }}
+                  key: {{ .usernameKey }}
+            - name: PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .secretName }}
+                  key: {{ .passwordKey }}
```

Comment:
My bad if I'm reading this wrong, but should all of these secrets refer to the same `.secretName` and `.passwordKey`?

Reply:
The secret name contains the key and both could be any value.

Reply:
Okay. If one of y'all could walk me through how that works at some point tomorrow, I'd v much appreciate it =)